In [22]:
import pandas as pd #pd is the alias
import numpy as np 
import datetime
from os import listdir 
from os.path import isfile, join
import glob 
import re

In [23]:
df = pd.read_csv("data-set/Mahamud Shakil - free_session_information.csv")
df

Time Stamp   Name of the candidate      Contact   \
0    8/24/2020 17:01:39  A.S.M. Raquibul Hasan   01722577922   
1    8/24/2020 10:08:54                Abdullah          NaN   
2     8/24/2020 9:20:28    Abdullah Bin Murad    01552443554   
3    7/28/2020 12:16:30       Abhijoy Mukherjee   7003206060   
4    7/10/2020 13:56:14     Abhipreeti Acharjee   8274839744   
..                  ...                     ...          ...   
963  1/22/2021 22:06:24       Ramanish Banerjee   8017179702   
964  1/22/2021 22:46:21          Jinia Banerjee   6289719175   
965  1/23/2021 10:57:08        Md Safiqul Islam   8250955315   
966  1/23/2021 12:27:32     Soumojit Chatterjee   8101726672   
967  1/23/2021 17:37:35  Debanjan Bhattacharjee   6290909245   

                               Email id   \
0                 raquibul.nsu@gmail.com   
1              sharif3abdullah@gmail.com   
2                   abmrifat54@gmail.com   
3           abhijoymukherjee01@gmail.com   
4               deepamandal007@gmail.com   
..                                   ...   
963           banerjeeramanish@gmail.com   
964          jiniabanerjee9727@gmail.com   
965           safiqulislam6006@gmail.com   
966                 imsoumojit@yahoo.com   
967  debanjanbhattacharjee2002@gmail.com   

                                         College name             Brach   \
0                     United International University             others   
1                                                  IIT       Engineering   
2       Shahjalal University of Science and Technology       Engineering   
3                                      Asutosh College        Statistics   
4    DEPARTMENT OF PURE MATHEMATICS, UNIVERSITY OF ...  PURE MATHEMATICS   
..                                                 ...               ...   
963                               MAULANA AZAD COLLEGE  Honors (BSC/MSC)   
964                               Maulona Azad College  Honors (BSC/MSC)   
965                               Maulana Azad College  Honors (BSC/MSC)   
966                               Maulana Azad College  Honors (BSC/MSC)   
967                               Maulana Azad College  Honors (BSC/MSC)   

    People having laptop  Souce of communication   Unnamed: 8  Unnamed: 9  \
0                     Yes               Facebook          NaN         NaN   
1                     NaN                     NaN         NaN         NaN   
2                     Yes                     NaN         NaN         NaN   
3                     Yes                  Friend         4.0         NaN   
4                     Yes                     NaN         2.0         NaN   
..                    ...                     ...         ...         ...   
963                   Yes                     NaN         NaN         NaN   
964                    No                     NaN         NaN         NaN   
965                   Yes                     NaN         NaN         NaN   
966                   Yes                     NaN         NaN         NaN   
967                   Yes                     NaN         NaN         NaN   

       Country   
0    Bangladesh  
1    Bangladesh  
2    Bangladesh  
3         India  
4         India  
..          ...  
963         NaN  
964         NaN  
965         NaN  
966         NaN  
967         NaN  

[968 rows x 11 columns]

In [24]:
df.shape

(968, 11)

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 968 entries, 0 to 967
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Time Stamp               948 non-null    object 
 1   Name of the candidate    947 non-null    object 
 2   Contact                  909 non-null    object 
 3   Email id                 946 non-null    object 
 4   College name             940 non-null    object 
 5   Brach                    933 non-null    object 
 6   People having laptop     940 non-null    object 
 7   Souce of communication   432 non-null    object 
 8   Unnamed: 8               318 non-null    float64
 9   Unnamed: 9               0 non-null      float64
 10  Country                  476 non-null    object 
dtypes: float64(2), object(9)
memory usage: 83.3+ KB


In [83]:
#Handling Missing Data 
df.isna().sum()

Timestamp      20
Name           21
Contact        59
Email          22
College        28
Department     35
HasLaptop      28
Source        536
Country       492
dtype: int64

# Dropping the Unnamed columns 

In [26]:
df.drop(['Unnamed: 8', 'Unnamed: 9'], axis = 1, inplace = True)

In [27]:
df

Time Stamp   Name of the candidate      Contact   \
0    8/24/2020 17:01:39  A.S.M. Raquibul Hasan   01722577922   
1    8/24/2020 10:08:54                Abdullah          NaN   
2     8/24/2020 9:20:28    Abdullah Bin Murad    01552443554   
3    7/28/2020 12:16:30       Abhijoy Mukherjee   7003206060   
4    7/10/2020 13:56:14     Abhipreeti Acharjee   8274839744   
..                  ...                     ...          ...   
963  1/22/2021 22:06:24       Ramanish Banerjee   8017179702   
964  1/22/2021 22:46:21          Jinia Banerjee   6289719175   
965  1/23/2021 10:57:08        Md Safiqul Islam   8250955315   
966  1/23/2021 12:27:32     Soumojit Chatterjee   8101726672   
967  1/23/2021 17:37:35  Debanjan Bhattacharjee   6290909245   

                               Email id   \
0                 raquibul.nsu@gmail.com   
1              sharif3abdullah@gmail.com   
2                   abmrifat54@gmail.com   
3           abhijoymukherjee01@gmail.com   
4               deepamandal007@gmail.com   
..                                   ...   
963           banerjeeramanish@gmail.com   
964          jiniabanerjee9727@gmail.com   
965           safiqulislam6006@gmail.com   
966                 imsoumojit@yahoo.com   
967  debanjanbhattacharjee2002@gmail.com   

                                         College name             Brach   \
0                     United International University             others   
1                                                  IIT       Engineering   
2       Shahjalal University of Science and Technology       Engineering   
3                                      Asutosh College        Statistics   
4    DEPARTMENT OF PURE MATHEMATICS, UNIVERSITY OF ...  PURE MATHEMATICS   
..                                                 ...               ...   
963                               MAULANA AZAD COLLEGE  Honors (BSC/MSC)   
964                               Maulona Azad College  Honors (BSC/MSC)   
965                               Maulana Azad College  Honors (BSC/MSC)   
966                               Maulana Azad College  Honors (BSC/MSC)   
967                               Maulana Azad College  Honors (BSC/MSC)   

    People having laptop  Souce of communication     Country   
0                     Yes               Facebook   Bangladesh  
1                     NaN                     NaN  Bangladesh  
2                     Yes                     NaN  Bangladesh  
3                     Yes                  Friend       India  
4                     Yes                     NaN       India  
..                    ...                     ...         ...  
963                   Yes                     NaN         NaN  
964                    No                     NaN         NaN  
965                   Yes                     NaN         NaN  
966                   Yes                     NaN         NaN  
967                   Yes                     NaN         NaN  

[968 rows x 9 columns]

# Update the columns name

In [34]:
df.rename(columns={"Time Stamp ": "Timestamp", "Name of the candidate ": "Name","Contact ":"Contact","Email id ": "Email","College name ":"College","Brach ":"Department","People having laptop ":"HasLaptop","Souce of communication ":"Source","Country ":"Country"},inplace=True)

In [35]:
df.columns

Index(['Timestamp', 'Name', 'Contact', 'Email', 'College', 'Department',
       'HasLaptop', 'Source', 'Country'],
      dtype='object')

In [36]:
df

Timestamp                    Name      Contact  \
0    8/24/2020 17:01:39  A.S.M. Raquibul Hasan   01722577922   
1    8/24/2020 10:08:54                Abdullah          NaN   
2     8/24/2020 9:20:28    Abdullah Bin Murad    01552443554   
3    7/28/2020 12:16:30       Abhijoy Mukherjee   7003206060   
4    7/10/2020 13:56:14     Abhipreeti Acharjee   8274839744   
..                  ...                     ...          ...   
963  1/22/2021 22:06:24       Ramanish Banerjee   8017179702   
964  1/22/2021 22:46:21          Jinia Banerjee   6289719175   
965  1/23/2021 10:57:08        Md Safiqul Islam   8250955315   
966  1/23/2021 12:27:32     Soumojit Chatterjee   8101726672   
967  1/23/2021 17:37:35  Debanjan Bhattacharjee   6290909245   

                                   Email  \
0                 raquibul.nsu@gmail.com   
1              sharif3abdullah@gmail.com   
2                   abmrifat54@gmail.com   
3           abhijoymukherjee01@gmail.com   
4               deepamandal007@gmail.com   
..                                   ...   
963           banerjeeramanish@gmail.com   
964          jiniabanerjee9727@gmail.com   
965           safiqulislam6006@gmail.com   
966                 imsoumojit@yahoo.com   
967  debanjanbhattacharjee2002@gmail.com   

                                               College        Department  \
0                     United International University             others   
1                                                  IIT       Engineering   
2       Shahjalal University of Science and Technology       Engineering   
3                                      Asutosh College        Statistics   
4    DEPARTMENT OF PURE MATHEMATICS, UNIVERSITY OF ...  PURE MATHEMATICS   
..                                                 ...               ...   
963                               MAULANA AZAD COLLEGE  Honors (BSC/MSC)   
964                               Maulona Azad College  Honors (BSC/MSC)   
965                               Maulana Azad College  Honors (BSC/MSC)   
966                               Maulana Azad College  Honors (BSC/MSC)   
967                               Maulana Azad College  Honors (BSC/MSC)   

    HasLaptop     Source     Country  
0         Yes  Facebook   Bangladesh  
1         NaN        NaN  Bangladesh  
2         Yes        NaN  Bangladesh  
3         Yes     Friend       India  
4         Yes        NaN       India  
..        ...        ...         ...  
963       Yes        NaN         NaN  
964        No        NaN         NaN  
965       Yes        NaN         NaN  
966       Yes        NaN         NaN  
967       Yes        NaN         NaN  

[968 rows x 9 columns]

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 968 entries, 0 to 967
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Timestamp   948 non-null    object
 1   Name        947 non-null    object
 2   Contact     909 non-null    object
 3   Email       946 non-null    object
 4   College     940 non-null    object
 5   Department  933 non-null    object
 6   HasLaptop   940 non-null    object
 7   Source      432 non-null    object
 8   Country     476 non-null    object
dtypes: object(9)
memory usage: 68.2+ KB


# Saving the datset after removing unnecessary things

In [ ]:
df.to_csv('free_session_information.csv')

# Batch information.csv dataset information

In [137]:
df2 = pd.read_csv("data-set/Mahamud Shakil - batch_inforamtion.csv")
df2

Time                  Email Address  \
0          8/1/20 23:05         deepanjan325@gmail.com   
1          8/1/20 23:06     ranubanerjee1984@gmail.com   
2          8/1/20 23:08         sharmisthaad@gmail.com   
3          8/1/20 23:10          saikat20171@gmail.com   
4          8/1/20 23:10       rahulroystat36@gmail.com   
..                  ...                            ...   
160  8/28/2020 12:15:37         souravkr2025@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  \
0           Deepanjan Biswas   
1              RANU BANERJEE   
2        Sharmistha Adhikari   
3               Saikat Mitra   
4                  Rahul Roy   
..                       ...   
160         Sourav Kumar Dan   
161         Srijani Adhikary   
162  Suvargha Ghosh Dastidar   
163                Titas Dey   
164             Trisha Dutta   

     How well you guys are understanding the covered topics ?  \
0                                                  5.0          
1                                                  5.0          
2                                                  1.0          
3                                                  4.0          
4                                                  5.0          
..                                                 ...          
160                                                3.0          
161                                                5.0          
162                                                4.0          
163                                                5.0          
164                                                4.0          

     Rate our way of training  How well your queries are getting solved   \
0                         5.0                                        5.0   
1                         5.0                                        5.0   
2                         1.0                                        1.0   
3                         5.0                                        5.0   
4                         5.0                                        5.0   
..                        ...                                        ...   
160                       4.0                                        4.0   
161                       5.0                                        5.0   
162                       5.0                                        5.0   
163                       5.0                                        5.0   
164                       4.0                                        5.0   

    Would you like to join our full time course ?  \
0                                             Yes   
1                                           Maybe   
2                                           Maybe   
3                                           Maybe   
4                                             Yes   
..                                            ...   
160                                           Yes   
161                                         Maybe   
162                                         Maybe   
163                                         Maybe   
164                                           Yes   

    Would you like to recommend this course to others ?  \
0                                                  Yes    
1                                                  Yes    
2                                                  Yes    
3                                                  Yes    
4                                                  Yes    
..                                                 ...    
160                                                NaN    
161                                                NaN    
162                                                NaN    
163             

In [138]:
df2.shape

(165, 13)

In [139]:
df2.columns

Index([' Time', 'Email Address', 'Name',
       'How well you guys are understanding the covered topics ?',
       'Rate our way of training', 'How well your queries are getting solved ',
       'Would you like to join our full time course ?',
       'Would you like to recommend this course to others ?',
       'Would you like to apply for paid internship programme by Softanbees  ?',
       'Your suggestion for improving the training programme (Any other queries)',
       'Please describe Here', 'Please describe here',
       'Would you like to recommend this course to others ?.1'],
      dtype='object')

In [140]:
df2[df2['Please describe Here'].isnull()]

Time                  Email Address  \
0          8/1/20 23:05         deepanjan325@gmail.com   
2          8/1/20 23:08         sharmisthaad@gmail.com   
3          8/1/20 23:10          saikat20171@gmail.com   
4          8/1/20 23:10       rahulroystat36@gmail.com   
5          8/1/20 23:10            as1973240@gmail.com   
..                  ...                            ...   
160  8/28/2020 12:15:37         souravkr2025@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  \
0           Deepanjan Biswas   
2        Sharmistha Adhikari   
3               Saikat Mitra   
4                  Rahul Roy   
5             Abhishek Singh   
..                       ...   
160         Sourav Kumar Dan   
161         Srijani Adhikary   
162  Suvargha Ghosh Dastidar   
163                Titas Dey   
164             Trisha Dutta   

     How well you guys are understanding the covered topics ?  \
0                                                  5.0          
2                                                  1.0          
3                                                  4.0          
4                                                  5.0          
5                                                  3.0          
..                                                 ...          
160                                                3.0          
161                                                5.0          
162                                                4.0          
163                                                5.0          
164                                                4.0          

     Rate our way of training  How well your queries are getting solved   \
0                         5.0                                        5.0   
2                         1.0                                        1.0   
3                         5.0                                        5.0   
4                         5.0                                        5.0   
5                         3.0                                        3.0   
..                        ...                                        ...   
160                       4.0                                        4.0   
161                       5.0                                        5.0   
162                       5.0                                        5.0   
163                       5.0                                        5.0   
164                       4.0                                        5.0   

    Would you like to join our full time course ?  \
0                                             Yes   
2                                           Maybe   
3                                           Maybe   
4                                             Yes   
5                                           Maybe   
..                                            ...   
160                                           Yes   
161                                         Maybe   
162                                         Maybe   
163                                         Maybe   
164                                           Yes   

    Would you like to recommend this course to others ?  \
0                                                  Yes    
2                                                  Yes    
3                                                  Yes    
4                                                  Yes    
5                                                Maybe    
..                                                 ...    
160                                                NaN    
161                                                NaN    
162                                                NaN    
163             

In [141]:
df2[df2['Please describe here'].isnull()]

Time                  Email Address  \
0          8/1/20 23:05         deepanjan325@gmail.com   
1          8/1/20 23:06     ranubanerjee1984@gmail.com   
2          8/1/20 23:08         sharmisthaad@gmail.com   
3          8/1/20 23:10          saikat20171@gmail.com   
4          8/1/20 23:10       rahulroystat36@gmail.com   
..                  ...                            ...   
159   8/28/2020 8:22:27     sonalisarenmumpi@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  \
0           Deepanjan Biswas   
1              RANU BANERJEE   
2        Sharmistha Adhikari   
3               Saikat Mitra   
4                  Rahul Roy   
..                       ...   
159             SONALI SAREN   
161         Srijani Adhikary   
162  Suvargha Ghosh Dastidar   
163                Titas Dey   
164             Trisha Dutta   

     How well you guys are understanding the covered topics ?  \
0                                                  5.0          
1                                                  5.0          
2                                                  1.0          
3                                                  4.0          
4                                                  5.0          
..                                                 ...          
159                                                5.0          
161                                                5.0          
162                                                4.0          
163                                                5.0          
164                                                4.0          

     Rate our way of training  How well your queries are getting solved   \
0                         5.0                                        5.0   
1                         5.0                                        5.0   
2                         1.0                                        1.0   
3                         5.0                                        5.0   
4                         5.0                                        5.0   
..                        ...                                        ...   
159                       5.0                                        5.0   
161                       5.0                                        5.0   
162                       5.0                                        5.0   
163                       5.0                                        5.0   
164                       4.0                                        5.0   

    Would you like to join our full time course ?  \
0                                             Yes   
1                                           Maybe   
2                                           Maybe   
3                                           Maybe   
4                                             Yes   
..                                            ...   
159                                         Maybe   
161                                         Maybe   
162                                         Maybe   
163                                         Maybe   
164                                           Yes   

    Would you like to recommend this course to others ?  \
0                                                  Yes    
1                                                  Yes    
2                                                  Yes    
3                                                  Yes    
4                                                  Yes    
..                                                 ...    
159                                                NaN    
161                                                NaN    
162                                                NaN    
163             

|Please describe Here, Please describe Here| with this name has two columns and also from total 165 rows null values exits 131 rows and another one is full of null values. so this two columns will not be very necessary for us.

# Delete unnecessary column/column with irrelevant data/no data at all

In [142]:
df2.drop(['Please describe Here',], axis = 1, inplace = True)

In [143]:
df2.drop(['Please describe here',], axis = 1, inplace = True)

In [144]:
df2

Time                  Email Address  \
0          8/1/20 23:05         deepanjan325@gmail.com   
1          8/1/20 23:06     ranubanerjee1984@gmail.com   
2          8/1/20 23:08         sharmisthaad@gmail.com   
3          8/1/20 23:10          saikat20171@gmail.com   
4          8/1/20 23:10       rahulroystat36@gmail.com   
..                  ...                            ...   
160  8/28/2020 12:15:37         souravkr2025@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  \
0           Deepanjan Biswas   
1              RANU BANERJEE   
2        Sharmistha Adhikari   
3               Saikat Mitra   
4                  Rahul Roy   
..                       ...   
160         Sourav Kumar Dan   
161         Srijani Adhikary   
162  Suvargha Ghosh Dastidar   
163                Titas Dey   
164             Trisha Dutta   

     How well you guys are understanding the covered topics ?  \
0                                                  5.0          
1                                                  5.0          
2                                                  1.0          
3                                                  4.0          
4                                                  5.0          
..                                                 ...          
160                                                3.0          
161                                                5.0          
162                                                4.0          
163                                                5.0          
164                                                4.0          

     Rate our way of training  How well your queries are getting solved   \
0                         5.0                                        5.0   
1                         5.0                                        5.0   
2                         1.0                                        1.0   
3                         5.0                                        5.0   
4                         5.0                                        5.0   
..                        ...                                        ...   
160                       4.0                                        4.0   
161                       5.0                                        5.0   
162                       5.0                                        5.0   
163                       5.0                                        5.0   
164                       4.0                                        5.0   

    Would you like to join our full time course ?  \
0                                             Yes   
1                                           Maybe   
2                                           Maybe   
3                                           Maybe   
4                                             Yes   
..                                            ...   
160                                           Yes   
161                                         Maybe   
162                                         Maybe   
163                                         Maybe   
164                                           Yes   

    Would you like to recommend this course to others ?  \
0                                                  Yes    
1                                                  Yes    
2                                                  Yes    
3                                                  Yes    
4                                                  Yes    
..                                                 ...    
160                                                NaN    
161                                                NaN    
162                                                NaN    
163             

# Update the columns name of Batch_information

In [145]:
df2.rename(columns={" Time": "Timestamp", "Email Address": "Email","How well you guys are understanding the covered topics ?": "UderstandingTopic(0-5)","Rate our way of training":"TrainingRating(0-5)","How well your queries are getting solved ":"Student_queries_sloved(0-5)","Would you like to join our full time course ?":"want_join_full_time_courses","Would you like to recommend this course to others ?":"recommend_course","Would you like to apply for paid internship programme by Softanbees  ?":"want_apply_paid_intership","Your suggestion for improving the training programme (Any other queries)":"suggestions_improving_program"},inplace=True)

In [146]:
df2.columns

Index(['Timestamp', 'Email', 'Name', 'UderstandingTopic(0-5)',
       'TrainingRating(0-5)', 'Student_queries_sloved(0-5)',
       'want_join_full_time_courses', 'recommend_course',
       'want_apply_paid_intership', 'suggestions_improving_program',
       'Would you like to recommend this course to others ?.1'],
      dtype='object')

In [147]:
df2.head()

Timestamp                       Email                 Name  \
0  8/1/20 23:05      deepanjan325@gmail.com     Deepanjan Biswas   
1  8/1/20 23:06  ranubanerjee1984@gmail.com        RANU BANERJEE   
2  8/1/20 23:08      sharmisthaad@gmail.com  Sharmistha Adhikari   
3  8/1/20 23:10       saikat20171@gmail.com         Saikat Mitra   
4  8/1/20 23:10    rahulroystat36@gmail.com            Rahul Roy   

   UderstandingTopic(0-5)  TrainingRating(0-5)  Student_queries_sloved(0-5)  \
0                     5.0                  5.0                          5.0   
1                     5.0                  5.0                          5.0   
2                     1.0                  1.0                          1.0   
3                     4.0                  5.0                          5.0   
4                     5.0                  5.0                          5.0   

  want_join_full_time_courses recommend_course want_apply_paid_intership  \
0                         Yes              Yes                       Yes   
1                       Maybe              Yes                     Maybe   
2                       Maybe              Yes                     Maybe   
3                       Maybe              Yes                       Yes   
4                         Yes              Yes                       Yes   

                       suggestions_improving_program  \
0                                                NaN   
1                                                 NO   
2                                                NaN   
3                                        All is well   
4  I want to know if you r going to tell us how t...   

  Would you like to recommend this course to others ?.1  
0                                                NaN     
1                                                NaN     
2                                                NaN     
3                                                NaN     
4                                                NaN

In [148]:
df2[df2['Would you like to recommend this course to others ?.1'].isnull()]

Timestamp                       Email                 Name  \
0    8/1/20 23:05      deepanjan325@gmail.com     Deepanjan Biswas   
1    8/1/20 23:06  ranubanerjee1984@gmail.com        RANU BANERJEE   
2    8/1/20 23:08      sharmisthaad@gmail.com  Sharmistha Adhikari   
3    8/1/20 23:10       saikat20171@gmail.com         Saikat Mitra   
4    8/1/20 23:10    rahulroystat36@gmail.com            Rahul Roy   
..            ...                         ...                  ...   
64   8/4/20 11:01  anupriyahalder99@gmail.com      Anupriya Halder   
65   8/4/20 12:32       sakshi90500@gmail.com        Kumari Sakshi   
66   8/4/20 17:50   sharmagautam990@gmail.com  Gautam Kumar Sharma   
67            NaN                         NaN                  NaN   
125           NaN                         NaN                  NaN   

     UderstandingTopic(0-5)  TrainingRating(0-5)  Student_queries_sloved(0-5)  \
0                       5.0                  5.0                          5.0   
1                       5.0                  5.0                          5.0   
2                       1.0                  1.0                          1.0   
3                       4.0                  5.0                          5.0   
4                       5.0                  5.0                          5.0   
..                      ...                  ...                          ...   
64                      4.0                  4.0                          4.0   
65                      5.0                  5.0                          5.0   
66                      4.0                  5.0                          4.0   
67                      NaN                  NaN                          NaN   
125                     NaN                  NaN                          NaN   

    want_join_full_time_courses recommend_course want_apply_paid_intership  \
0                           Yes              Yes                       Yes   
1                         Maybe              Yes                     Maybe   
2                         Maybe              Yes                     Maybe   
3                         Maybe              Yes                       Yes   
4                           Yes              Yes                       Yes   
..                          ...              ...                       ...   
64                           No              Yes                        No   
65                        Maybe              Yes                     Maybe   
66                          Yes              Yes                       Yes   
67                          NaN              NaN                       NaN   
125                         NaN              NaN                       NaN   

                         suggestions_improving_program  \
0                                                  NaN   
1                                                   NO   
2                                                  NaN   
3                                          All is well   
4    I want to know if you r going to tell us how t...   
..                                                 ...   
64                                                 NaN   
65   Tgere's no suggestion as it was teally nice an...   
66                                                  No   
67                                                 NaN   
125                                                NaN   

    Would you like to recommend this course to others ?.1  
0                                                  NaN     
1                                                  NaN     
2                                                  NaN     
3                                                  NaN     
4                                                  NaN     
..                                                 ...     
64                                                 NaN     
65                                                 NaN     
66                  

In [149]:
df2[df2['recommend_course'].isnull()]

Timestamp                          Email  \
67                  NaN                            NaN   
68         8/5/20 22:50        snehaguha2017@gmail.com   
69         8/5/20 22:53      mayuridharmegha@gmail.com   
70         8/5/20 22:55           srijanim75@gmail.com   
71         8/5/20 22:55  sanchitachowdhury02@gmail.com   
..                  ...                            ...   
160  8/28/2020 12:15:37         souravkr2025@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  UderstandingTopic(0-5)  TrainingRating(0-5)  \
67                       NaN                     NaN                  NaN   
68                Sneha Guha                     5.0                  5.0   
69               Mayuri Dhar                     5.0                  5.0   
70           Srijani Mondal                      5.0                  5.0   
71        Sanchita Chowdhury                     5.0                  5.0   
..                       ...                     ...                  ...   
160         Sourav Kumar Dan                     3.0                  4.0   
161         Srijani Adhikary                     5.0                  5.0   
162  Suvargha Ghosh Dastidar                     4.0                  5.0   
163                Titas Dey                     5.0                  5.0   
164             Trisha Dutta                     4.0                  4.0   

     Student_queries_sloved(0-5) want_join_full_time_courses recommend_course  \
67                           NaN                         NaN              NaN   
68                           5.0                       Maybe              NaN   
69                           5.0                          No              NaN   
70                           5.0                          No              NaN   
71                           5.0                          No              NaN   
..                           ...                         ...              ...   
160                          4.0                         Yes              NaN   
161                          5.0                       Maybe              NaN   
162                          5.0                       Maybe              NaN   
163                          5.0                       Maybe              NaN   
164                          5.0                         Yes              NaN   

    want_apply_paid_intership  \
67                        NaN   
68                         qt   
69                         No   
70                         No   
71                         No   
..                        ...   
160                        No   
161                     Maybe   
162                     Maybe   
163                     Maybe   
164                     Maybe   

                         suggestions_improving_program  \
67                                                 NaN   
68                                                 NaN   
69                                                 NaN   
70                                                 NaN   
71   It's really good...so for me there is nothing ...   
..                                                 ...   
160                                                NaN   
161                                                No    
162            The training programme was quite good.    
163                                                NaN   
164                           Training program is good   

    Would you like to recommend this course to others ?.1  
67                                                 NaN     
68                                                 Yes     
69                                                 Yes     
70                                                 Yes     
71                       

In [150]:
#Handling Missing Data 
df2.isna().sum()

Timestamp                                                 2
Email                                                     2
Name                                                      2
UderstandingTopic(0-5)                                    2
TrainingRating(0-5)                                       2
Student_queries_sloved(0-5)                               2
want_join_full_time_courses                               2
recommend_course                                         98
want_apply_paid_intership                                 2
suggestions_improving_program                            88
Would you like to recommend this course to others ?.1    69
dtype: int64

In [151]:
import math
def merge_recommend_columns(col1,col2):
    """
    if col2 value is NaN or null then the value will replace by the col1 value
    also if col1 value is not Nan
    """
    if col2 == "NaN" or col2 == " " and col1 != "NaN":
        col2 = col1
    return col2

In [152]:
df2['recommend1'] = df2.apply(lambda row : merge_recommend_columns(row['recommend_course'],row['Would you like to recommend this course to others ?.1']), axis = 1)

In [153]:
df2

Timestamp                          Email  \
0          8/1/20 23:05         deepanjan325@gmail.com   
1          8/1/20 23:06     ranubanerjee1984@gmail.com   
2          8/1/20 23:08         sharmisthaad@gmail.com   
3          8/1/20 23:10          saikat20171@gmail.com   
4          8/1/20 23:10       rahulroystat36@gmail.com   
..                  ...                            ...   
160  8/28/2020 12:15:37         souravkr2025@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  UderstandingTopic(0-5)  TrainingRating(0-5)  \
0           Deepanjan Biswas                     5.0                  5.0   
1              RANU BANERJEE                     5.0                  5.0   
2        Sharmistha Adhikari                     1.0                  1.0   
3               Saikat Mitra                     4.0                  5.0   
4                  Rahul Roy                     5.0                  5.0   
..                       ...                     ...                  ...   
160         Sourav Kumar Dan                     3.0                  4.0   
161         Srijani Adhikary                     5.0                  5.0   
162  Suvargha Ghosh Dastidar                     4.0                  5.0   
163                Titas Dey                     5.0                  5.0   
164             Trisha Dutta                     4.0                  4.0   

     Student_queries_sloved(0-5) want_join_full_time_courses recommend_course  \
0                            5.0                         Yes              Yes   
1                            5.0                       Maybe              Yes   
2                            1.0                       Maybe              Yes   
3                            5.0                       Maybe              Yes   
4                            5.0                         Yes              Yes   
..                           ...                         ...              ...   
160                          4.0                         Yes              NaN   
161                          5.0                       Maybe              NaN   
162                          5.0                       Maybe              NaN   
163                          5.0                       Maybe              NaN   
164                          5.0                         Yes              NaN   

    want_apply_paid_intership  \
0                         Yes   
1                       Maybe   
2                       Maybe   
3                         Yes   
4                         Yes   
..                        ...   
160                        No   
161                     Maybe   
162                     Maybe   
163                     Maybe   
164                     Maybe   

                         suggestions_improving_program  \
0                                                  NaN   
1                                                   NO   
2                                                  NaN   
3                                          All is well   
4    I want to know if you r going to tell us how t...   
..                                                 ...   
160                                                NaN   
161                                                No    
162            The training programme was quite good.    
163                                                NaN   
164                           Training program is good   

    Would you like to recommend this course to others ?.1 recommend1  
0                                                  NaN           NaN  
1                                                  NaN           NaN  
2                                                  NaN           NaN  
3                                        

In [154]:
#Handling Missing Data 
df2.isna().sum()

Timestamp                                                 2
Email                                                     2
Name                                                      2
UderstandingTopic(0-5)                                    2
TrainingRating(0-5)                                       2
Student_queries_sloved(0-5)                               2
want_join_full_time_courses                               2
recommend_course                                         98
want_apply_paid_intership                                 2
suggestions_improving_program                            88
Would you like to recommend this course to others ?.1    69
recommend1                                               69
dtype: int64

total recommend course null values 98 whereas  Would you like to recommend this course to others ?.1 also has 69 null values here build a function for this tow columns

In [116]:
help(merge_recommend_columns)

Help on function merge_recommend_columns in module __main__:

merge_recommend_columns(col1, col2)
    if col2 value is NaN or null then the value will replace by the col1 value
    also if col1 value is not Nan



In [155]:
df2.drop(['Would you like to recommend this course to others ?.1',], axis = 1, inplace = True)
df2.drop(['recommend1',], axis = 1, inplace = True)

In [156]:
df2

Timestamp                          Email  \
0          8/1/20 23:05         deepanjan325@gmail.com   
1          8/1/20 23:06     ranubanerjee1984@gmail.com   
2          8/1/20 23:08         sharmisthaad@gmail.com   
3          8/1/20 23:10          saikat20171@gmail.com   
4          8/1/20 23:10       rahulroystat36@gmail.com   
..                  ...                            ...   
160  8/28/2020 12:15:37         souravkr2025@gmail.com   
161  8/28/2020 20:45:00   srijani.adhikary13@gmail.com   
162  8/28/2020 20:45:07         suvargha2000@gmail.com   
163  8/28/2020 21:19:05         titutitasdey@gmail.com   
164  8/28/2020 21:58:42  trishadutta25042001@gmail.com   

                        Name  UderstandingTopic(0-5)  TrainingRating(0-5)  \
0           Deepanjan Biswas                     5.0                  5.0   
1              RANU BANERJEE                     5.0                  5.0   
2        Sharmistha Adhikari                     1.0                  1.0   
3               Saikat Mitra                     4.0                  5.0   
4                  Rahul Roy                     5.0                  5.0   
..                       ...                     ...                  ...   
160         Sourav Kumar Dan                     3.0                  4.0   
161         Srijani Adhikary                     5.0                  5.0   
162  Suvargha Ghosh Dastidar                     4.0                  5.0   
163                Titas Dey                     5.0                  5.0   
164             Trisha Dutta                     4.0                  4.0   

     Student_queries_sloved(0-5) want_join_full_time_courses recommend_course  \
0                            5.0                         Yes              Yes   
1                            5.0                       Maybe              Yes   
2                            1.0                       Maybe              Yes   
3                            5.0                       Maybe              Yes   
4                            5.0                         Yes              Yes   
..                           ...                         ...              ...   
160                          4.0                         Yes              NaN   
161                          5.0                       Maybe              NaN   
162                          5.0                       Maybe              NaN   
163                          5.0                       Maybe              NaN   
164                          5.0                         Yes              NaN   

    want_apply_paid_intership  \
0                         Yes   
1                       Maybe   
2                       Maybe   
3                         Yes   
4                         Yes   
..                        ...   
160                        No   
161                     Maybe   
162                     Maybe   
163                     Maybe   
164                     Maybe   

                         suggestions_improving_program  
0                                                  NaN  
1                                                   NO  
2                                                  NaN  
3                                          All is well  
4    I want to know if you r going to tell us how t...  
..                                                 ...  
160                                                NaN  
161                                                No   
162            The training programme was quite good.   
163                                                NaN  
164                           Training program is good  

[165 rows x 10 columns]

In [178]:
#convert the datatypes to category format 
df2['UderstandingTopic(0-5)'] = df2['UderstandingTopic(0-5)'].astype("category")
df2['TrainingRating(0-5)'] = df2['TrainingRating(0-5)'].astype("category")
df2['Student_queries_sloved(0-5)'] = df2['Student_queries_sloved(0-5)'].astype("category")

In [181]:
df2.dtypes

Timestamp                          object
Email                              object
Name                               object
UderstandingTopic(0-5)           category
TrainingRating(0-5)              category
Student_queries_sloved(0-5)      category
want_join_full_time_courses        object
recommend_course                   object
want_apply_paid_intership          object
suggestions_improving_program      object
dtype: object

# Saving the datset after removing unnecessary things

In [182]:
df2.shape

(165, 10)

In [183]:
df2.to_csv('batch_inforamtion.csv')